engine/clientdb/log.go

```diff
@@ -2,50 +2,126 @@ package clientdb
 
 import (
 	"log/slog"
-	"time"
 
 	"dagger.io/dagger/telemetry"
-	"go.opentelemetry.io/otel/log"
-	sdklog "go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
 	"go.opentelemetry.io/otel/trace"
 	otlpcommonv1 "go.opentelemetry.io/proto/otlp/common/v1"
+	otlplogsv1 "go.opentelemetry.io/proto/otlp/logs/v1"
+	otlpresourcev1 "go.opentelemetry.io/proto/otlp/resource/v1"
+	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
 )
 
-func (dbLog *Log) Record() sdklog.Record {
-	var rec sdklog.Record
-
-	rec.SetTimestamp(time.Unix(0, dbLog.Timestamp))
-
-	rec.SetSeverity(log.Severity(dbLog.Severity))
-
-	if dbLog.TraceID.Valid {
-		tid, _ := trace.TraceIDFromHex(dbLog.TraceID.String)
-		rec.SetTraceID(tid)
+func LogsToPB(dbLog []Log) []*otlplogsv1.ResourceLogs {
```

Comment:
NOTE: this converts from our SQLite database `Log` model, as opposed to the `[]sdklog.Record` input you might expect from normal OTLP transformations.

Previously we went from DB model -> `sdklog.Record` -> OTLP, now we just skip the middle step.